src/pyhf/contrib/utils.py

```diff
@@ -91,7 +91,12 @@ def download(archive_url, output_directory, force=False, compress=False):
                     with tarfile.open(
                         mode="r:*", fileobj=BytesIO(response.content)
                     ) as archive:
-                        archive.extractall(output_directory)
+                        # TODO: Simplify after pyhf is Python 3.12+ only
+                        # c.f. https://docs.python.org/3.12/library/tarfile.html#extraction-filters
+                        if hasattr(tarfile, "data_filter"):
+                            archive.extractall(output_directory, filter="data")
```

Comment:
This won't ever get run in the current tests and so Codecov is going to complain, but is passes on the testing Python 3.12 branch that I have and so will get covered in the future.